### PR TITLE
[RuntimeEnv] Support setting actor level env vars for Java worker

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
This PR supports setting actor level env vars for Java worker in runtime env.
General API looks like:
```java
RuntimeEnv runtimeEnv = new RuntimeEnv.Builder()
    .addEnvVar("KEY1", "A")
    .addEnvVar("KEY2", "B")
    .addEnvVar("KEY1", "C")  // This overwrites "KEY1" to "C"
    .build();

ActorHandle<A> actor1 = Ray.actor(A::new).setRuntimeEnv(runtimeEnv).remote();
```

If `num-java-workers-per-process` > 1, it will never reuse the worker process except they have the same runtime envs.

## Related issue number
#21463

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/test/src/main/java/io/ray/test/RuntimeEnvTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      return SystemUtil.pid();
+    }
+  }
+
+  public void testEnvironmentVariable() {
+    int pid1 = 0;
+    int pid2 = 0;
+    {
+      RuntimeEnv runtimeEnv =
+          new RuntimeEnvBuilder()
```
Comment (SongGuyang) [inline java/test/src/main/java/io/ray/test/RuntimeEnvTest.java:37]: Do we need to put `RuntimeEnvBuilder ` into `RuntimeEnv `? like:
```
RuntimeEnv runtimeEnv =
          new RuntimeEnv.RuntimeEnvBuilder()
```
I'm not sure. This two ways are all good for me. @raulchen what do you think?

**Review by SongGuyang — COMMENTED** [review]
Do you plan to add job level and task level API in current PR or some separate PRs?

**Comment by jovany-wang** [comment]
> Do you plan to add job level and task level API in current PR or some separate PRs?
Both are fine to me. Maybe it's better to merge this one first?

File: java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  @Override
+  public String toJsonBytes() {
+    Map<String, Object> json = new HashMap<>();
+    if (!envVars.isEmpty()) {
+      /// DO NOT hardcode this key.
+      json.put("env_vars", envVars);
+    }
+    return GSON.toJson(json);
```
[the remark this answers is not in the document]
Comment (jovany-wang) [reply java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java]: Addressed.

**Comment by jovany-wang** [comment]
@SongGuyang @raulchen Take more looks at the high level API?

**Comment by jovany-wang** [comment]
Hi @SongGuyang @raulchen , very thanks for your suggestions, and I moved `Builder` inner `RuntimeEnv` class. 

The API looks like:
```java
      RuntimeEnv runtimeEnv =
          new RuntimeEnv.Builder()
              .addEnvVar("KEY1", "A")
              .addEnvVar("KEY2", "B")
              .addEnvVar("KEY1", "C")
              .build();
```

Feel free to comment more suggestions, thanks!

**Review by SongGuyang — APPROVED** [review]
LGTM!

**Review by raulchen — APPROVED** [review]

**Pull request merged**